File: HomeBudget.Data/Entities/Budget.cs
namespace HomeBudget.Data.Entities;

public class Budget
{
    public int Id { get; set; }
    public decimal Amount { get; set; }
    public int Month { get; set; }
    public int Year { get; set; }

    public int UserId { get; set; }
    public int CategoryId { get; set; }
    public int? HouseholdId { get; set; }

    public User User { get; set; } = null!;
    public Category Category { get; set; } = null!;
    public Household? Household { get; set; }
}

File: HomeBudgetMobile/src/models/budget.ts
export interface BudgetDto {
  id: number;
  amount: number;
  spent: number;
  remaining: number;
  percentage: number;
  month: number;
  year: number;
  categoryName: string;
  categoryIcon: string;
  categoryColor: string;
  categoryId: number;
  userName: string;
  userId: number;
}

export interface CreateBudgetDto {
  amount: number;
  month: number;
  year: number;
  userId: number;
  categoryId: number;
  householdId: number;
}
